scarb/src/bin/scarb/commands/fmt.rs

```diff
@@ -3,10 +3,21 @@ use anyhow::Result;
 use crate::args::FmtArgs;
 use crate::errors::error_with_exit_code;
 use scarb::core::Config;
-use scarb::ops;
+use scarb::ops::{self, FmtAction};
 
 #[tracing::instrument(skip_all, level = "info")]
 pub fn run(args: FmtArgs, config: &Config) -> Result<()> {
+    // The action the formatted should perform,
+    // e.g. check formatting, format in place, or emit formatted file to stdout.
+    let action = if args.check {
```

Comment:
this is meant to allow people to do `scarb fmt --check`, `scarb fmt --emit stdout` as well as `scarb fmt --check --emit stdout` and still obtain coherent results (check has priority over emit)

Reply:
Why not disallow using both `--check` and `--emit` simultaneously in clap then?

Reply:
I don't know how to do it, but I'll investigate!

Reply:
You can take a look at https://docs.rs/clap/latest/clap/struct.Arg.html#method.conflicts_with